Naming an Array 'new_array' causes a C error

**V version:** 
0.1.26 83289d7.f91e3a5
**OS:** 
windows 10
**What did you do?**
mut new_array:=[]string
new_array<<"hello"

**What did you expect to see?**
The program getting compiled

**What did you see instead?**
```
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c: In function 'wmain':
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c:4665:27: error: called object 'new_array' is not a function or function pointer
  array_string new_array = new_array(0, 0, sizeof(string));
                           ^~~~~~~~~
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c:4665:15: note: declared here
  array_string new_array = new_array(0, 0, sizeof(string));
               ^~~~~~~~~
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c: In function 'strconv__ftoa__f32_to_str':
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c:5032:8: warning: initialized field overwritten [-Woverride-init]
   .u = 0,
        ^
C:\Users\malt2\AppData\Local\Temp\v\test.tmp.c:5032:8: note: (near initialization for '(anonymous).u')
...
==================
(Use `v -cg` to print the entire error message)

builder error: C error.
```

--- a/vlib/builtin/array.v
+++ b/vlib/builtin/array.v
@@ -28,6 +28,10 @@ fn new_array(mylen int, cap int, elm_size int) array {
 	return arr
 }
 
+fn __new_array(mylen int, cap int, elm_size int) array {
+	return new_array(mylen, cap, elm_size)
+}
+
 // TODO
 pub fn make(len int, cap int, elm_size int) array {
 	return new_array(len, cap, elm_size)

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -865,7 +865,8 @@ fn (g mut Gen) expr(node ast.Expr) {
 				elem_sym := g.table.get_type_symbol(it.elem_type)
 				elem_type_str := g.typ(it.elem_type)
 				if it.exprs.len == 0 {
-					g.write('new_array($it.exprs.len, $it.exprs.len, sizeof($elem_type_str))')
+					// use __new_array to fix conflicts when the name of the variable is new_array
+					g.write('__new_array($it.exprs.len, $it.exprs.len, sizeof($elem_type_str))')
 				} else {
 					len := it.exprs.len
 					g.write('new_array_from_c_array($len, $len, sizeof($elem_type_str), ')

